areas/extension/tests/AzureMcp.Extension.UnitTests/AzdCommandTest.cs

```diff
@@ -0,0 +1,342 @@
+                    Arg.Any<int>());
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable("PATH", originalPath);
+                AzdCommand.ClearCachedAzdPath();
+            }
+        }
+        finally
+        {
+            CleanupTempDirectory(tempDir);
+        }
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_ReturnsErrorResponse_WhenCommandFails()
+    {
+        // Arrange
+        var command = new AzdCommand(_logger);
+        var parser = new Parser(command.GetCommand());
+        var args = parser.Parse("--command \"env invalid-command\" --cwd test-dir");
+        var context = new CommandContext(_serviceProvider);
+
+        var errorMessage = "Error: azd env: 'invalid-command' is not an azd command.";
+
+        _processService.ExecuteAsync(
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<int>())
+            .Returns(new ProcessResult(1, string.Empty, errorMessage, "azd env invalid-command --cwd test-dir --no-prompt"));
+
+        var tempDir = CreateTempAzdCliDirectory();
+
+        try
+        {
+            var azdPath = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+                ? Path.Combine(tempDir, "azd.exe")
+                : Path.Combine(tempDir, "azd");
+
+            File.WriteAllText(azdPath, "mock azd executable");
+
+            var originalPath = Environment.GetEnvironmentVariable("PATH");
+            try
+            {
+                Environment.SetEnvironmentVariable("PATH", tempDir + Path.PathSeparator + originalPath);
+                AzdCommand.ClearCachedAzdPath();
+
+                // Act
+                var response = await command.ExecuteAsync(context, args);
+
+                // Assert
+                Assert.NotNull(response);
+                Assert.Equal(500, response.Status);
+                Assert.Equal(errorMessage, response.Message);
+            }
+            finally
+            {
+                Environment.SetEnvironmentVariable("PATH", originalPath);
+                AzdCommand.ClearCachedAzdPath();
+            }
+        }
+        finally
+        {
+            CleanupTempDirectory(tempDir);
+        }
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_HandlesException_AndSetsException()
+    {
+        // Arrange
+        var command = new AzdCommand(_logger);
+        var parser = new Parser(command.GetCommand());
+        var args = parser.Parse("--command \"env list\" --cwd test-dir");
+        var context = new CommandContext(_serviceProvider);
+
+        var exceptionMessage = "Azure Developer CLI executable not found";
+
+        _processService.ExecuteAsync(
+            Arg.Any<string>(),
+            Arg.Any<string>(),
+            Arg.Any<int>())
+            .Returns(Task.FromException<ProcessResult>(new FileNotFoundException(exceptionMessage)));
```

Comment:
`ThrowsAsync()` is less verbose :)
```suggestion
            .ThrowsAsync(new FileNotFoundException(exceptionMessage));
```

Reply:
Changed the line to include .ThrowsAsync